File: backend/application-service/src/main/java/com/recruitment/platform/application/model/ApplicationNote.java
package com.recruitment.platform.application.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Entity
@Getter
@Setter
@Table(name = "application_notes")
public class ApplicationNote {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "application_id", nullable = false)
    private Long applicationId;

    @Column(name = "author_user_id", nullable = false)
    private Long authorUserId;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();
}

File: frontend/recruitment_platform_app/lib/models/application_note.dart
class ApplicationNote {
  final int id;
  final int applicationId;
  final int authorUserId;
  final String content;
  final DateTime createdAt;

  ApplicationNote({
    required this.id,
    required this.applicationId,
    required this.authorUserId,
    required this.content,
    required this.createdAt,
  });

  factory ApplicationNote.fromJson(Map<String, dynamic> json) {
    return ApplicationNote(
      id: json['id'],
      applicationId: json['applicationId'],
      authorUserId: json['authorUserId'],
      content: json['content'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}
